### ISSUE TITLE
Questions on Run() function in marmitek-gw/marmitek/mochad_reader.py

### ISSUE THREAD

**Issue by LaureCremieux** [pawmint/marmitek-gw#3]
# Run() function in marmitek-gw/marmitek/mochad_reader.py

There's one thing I would change in the last part of the function. Here it starts:

```
def run(timezone): 
```

Create an empty dict aimed to be filled with (sensor: lastvalue):
```
    lastDoorEvents = {}
```

Take raw signals extracted from socket and return data = formatted signal (type, sensor, sensorkind, value, date)
```    
for lines in read_from_mochad(): 
        for signal in lines[:-1]:   
            logger.debug('Signal received: %s' % signal)  
            data = gather_data(signal, timezone) 
            if data is None: 
                continue 
```
And here comes the part I find weird: the filter to avoid getting two identical signals in a row from the door sensors (2xOFF):

```
            if(data['sensorKind'] != 'door' 
             or lastDoorEvents.get(data['sensor'], "") != data['value']): 
                lastDoorEvents[data['sensor']] = data['value']  
                yield data
```
So basically:
* if data comes from motion sensor → Why? 
* or if two consecutive signals are different 
then register value in lastDoorEvents

I would rather write:
* if data comes from door sensor 
* and if two consecutive signals are different 
then register value in lastDoorEvents

```
            if(data['sensorKind'] == 'door'
             and lastDoorEvents.get(data['sensor'], "") != data['value']): 
                lastDoorEvents[data['sensor']] = data['value'] 
                yield data
```

Voilà, dis moi!

**Comment by RomainEndelin** [comment]
Okay, I understand now!
That code actually kind of works, but still no: it sucks !

The trick to understand is that this `if` is not for filtering the data, but rather to yield them, so the condition has to be the negation of the intuitive condition for filtering.
But it sucks for 2 reasons:

* Given a112233 (door sensor) and A1 (motion sensor), if we have a112233-Normal -> A1-OFF -> A112233-Normal, the second door sensor message won't be filtered because A1 was in the middle.
* It's way to complicated anyway !
